tests/unit/api/main.py

```diff
@@ -412,6 +412,26 @@ def test_node_register(self):
                   "password": "tapeworm"})
         api._must_find(model.Node, 'node-99')
 
+    def test_node_register_with_metadata(self):
+        api.node_register('node-99', obm={
+                  "type": "http://schema.massopencloud.org/haas/v0/obm/ipmi",
+                  "host": "ipmihost",
+                  "user": "root",
+                  "password": "tapeworm"}, metadata={
+                  "EK": "pk"})
```

Comment:
Similar point to the thing I mentioned in rest_api.md; in general, data structures that are sub-components should be indented further.